inst/fit_model/fit_model.cpp

```diff
@@ -222,37 +222,48 @@ int main(int argc, char *argv[]) {
                                                art15plus_num,
                                                art15plus_isperc};
 
-  // Fit the model
-  leapfrog::State<double> state_current(params.age_groups_pop, params.num_genders,
-                                        params.disease_stages, params.age_groups_hiv,
-                                        params.treatment_stages);
-
-  // TODO: Make this runnable without passing intermediate data in
-  leapfrog::internal::initialise_model_state(params, state_current);
-  auto state_next = state_current;
   leapfrog::internal::IntermediateData<double> intermediate(params.age_groups_pop, params.age_groups_hiv,
                                                             params.num_genders,
                                                             params.disease_stages, params.treatment_stages,
                                                             params.age_groups_hiv_15plus);
-  intermediate.reset();
+  leapfrog::State<double> state_current(params.age_groups_pop, params.num_genders,
+                                        params.disease_stages, params.age_groups_hiv,
+                                        params.treatment_stages);
 
   std::vector<int> save_steps(61);
   std::iota(save_steps.begin(), save_steps.end(), 0);
   leapfrog::StateSaver<double> state_output(sim_years, save_steps, params.age_groups_pop, params.num_genders,
                                             params.disease_stages, params.age_groups_hiv,
                                             params.treatment_stages);
-  // Save initial state
-  state_output.save_state(state_current, 0);
-
-  // Each time step is mid-point of the year
-  for (int step = 1; step <= sim_years; ++step) {
-    state_next.reset();
-    leapfrog::run_general_pop_demographic_projection(step, params, state_current, state_next, intermediate);
-    leapfrog::run_hiv_pop_demographic_projection(step, params, state_current, state_next, intermediate);
-    leapfrog::run_hiv_model_simulation(step, params, state_current, state_next, intermediate);
-    state_output.save_state(state_next, step);
-    std::swap(state_current, state_next);
-    intermediate.reset();
+
+  const char *is_profiling = std::getenv("PROFILING");
```

Comment:
why not pass this (n) as an arg with default 0?

Reply:
Yes I don't know! No good reason, let me update that